### ISSUE TITLE
DeepSpeed Megatron [BUG]

### ISSUE THREAD

**Issue by koliaok** [microsoft/DeepSpeed#2417]
Hi, I tried Finetuning a Pytorch XLM roberta xl 3B model with Deepspeed. And I changed the megatron ML code to the XLM roberta pretraining code, uploaded the XLM-roberta xl pre-training model, and performed fine-tuning.

I have a question here.
1) The original Huggingface XLM-roberta xl was loaded as is, and 2) megatron ml (Megatron-LM-v1.1.5-ZeRO3 exampl [https://github.com/microsoft/DeepSpeedExamples/tree/master/megatron/Megatron- LM-v1.1.5-ZeRO3]) of transformers.py pre-training code and uploading the model by changing the pre-training code. Hyperparameters such as the default optimizer are the same.

Case 2 has much lower accuracy than case 1. Even if I change the Megatron ML transformers.py code to XLM-roberta xl and load the model, does it not produce the same result because of other reasons such as Model Parallelism logic, kernel fusion, parameter initialize, etc. in the code?

Can you tell me what is causing the two cases to not produce similar results?

**Comment by GuanhuaWang** [comment]
Hi @koliaok thx for finding this issue.

Regarding to the `DeepSpeedExamples/megatron` folder, acturally we are not maintaining it anymore (details [here](https://github.com/microsoft/DeepSpeedExamples/blob/master/megatron/README.md)).

Could you try to reproduce with megatron-deepspeed repo (https://github.com/microsoft/Megatron-DeepSpeed) and let us know if this issue still exists?

**Comment by koliaok** [comment]
Thank you @GuanhuaWang 
I try to reproduce megatron-deepspeed repo(https://github.com/microsoft/Megatron-DeepSpeed) !

**Comment by GuanhuaWang** [comment]
@koliaok , feel free to let us know if the problem still exists.

**Comment by jeffra** [comment]
@koliaok, please re-open if you are still seeing an issue in the new repo.
